### DISCUSSION TITLE
command execution fail if *some* wildcards can't expend. A warning is suffice

### DISCUSSION THREAD

**Discussion by shula** [fish-shell/fish-shell#8936] [General]
`ls *mp3 *ogg *wav *flac`

Current behavior:   If no "ogg" found, then **nothing** will be listed. Yes, it's in the docs.

But, in any other system, `*` means:  "zero or more".  Also in fish (the "4 exceptions": for, set, etc.)
What's the idea behind stopping execution altogether?

The top example should just show the files that do exist, not complain about what's missing.
Possible outcome:

```
fish> ls *mp3 *ogg
warning: *ogg not found  <<----- STDERR
Song1.mp3       <<----- STDOUT
Audio2.mp3     <<----- STDOUT
Cliplip.mp3      <<----- STDOUT
```

Otherwise, what's the workaround for listing multiple patterns files?
Isn't fish meant to be easier and more simple?

(on a personal tone: fish is nice, some things are annoying - but this ^^ one is crossing the line)

**Comment by faho** [comment]
If you want to have your globs expand to nothing, do what the docs say: Use them with `for`, `set`, `count` or a variable override:

```fish
set -l files *.mp3 *.ogg
ls $files
```

Maybe make it `set -q files[1] && ls $files` in there to only do anything if at least one file had been given.

Or

```fish
files={*.mp3,*.ogg} ls $files
```

for short.

We used to have just a warning, and that's the worst of both worlds. You get a warning you don't want *and* `ls` does something quite different from `ls foo.mp3` - it starts listing the entire current directory.

So we chose between not doing anything and erroring out, and we picked erroring out.
